samples/multiplatform/kmp-lib-sample/src/commonMain/kotlin/com/apollographql/apollo/kmpsample/data/ApolloCoroutinesRepository.kt

```diff
@@ -1,26 +1,39 @@
 package com.apollographql.apollo.kmpsample.data
 
-import com.apollographql.apollo.kmpsample.fragment.RepositoryFragment
+import com.apollographql.apollo.ApolloClient
 import com.apollographql.apollo.kmpsample.GithubRepositoriesQuery
 import com.apollographql.apollo.kmpsample.GithubRepositoryCommitsQuery
 import com.apollographql.apollo.kmpsample.GithubRepositoryDetailQuery
 import com.apollographql.apollo.kmpsample.fragment.RepositoryDetail
+import com.apollographql.apollo.kmpsample.fragment.RepositoryFragment
 import com.apollographql.apollo.kmpsample.type.OrderDirection
 import com.apollographql.apollo.kmpsample.type.PullRequestState
 import com.apollographql.apollo.kmpsample.type.RepositoryOrderField
+import com.apollographql.apollo.network.ApolloHttpNetworkTransport
+import kotlinx.coroutines.flow.single
 
 /**
  * An implementation of a [GitHubDataSource] that shows how we can use coroutines to make our apollo requests.
  */
-class ApolloCoroutinesRepository(private val service: ApolloCoroutinesService) {
+class ApolloCoroutinesRepository {
+  private val apolloClient = ApolloClient(
+      networkTransport = ApolloHttpNetworkTransport(
+          serverUrl = "https://api.github.com/graphql",
+          httpHeaders = mapOf(
+              "Accept" to "application/json",
+              "Content-Type" to "application/json",
+              "Authorization" to "bearer $GITHUB_KEY"
```

Comment:
A common use case is to have an interceptor that can refresh the token. Should we add a concept of interceptors there as well ?

Reply:
Sure, we can. I guess we can do this in next PR. As we have to make abstraction that works for both OkHttp and NSURLSession.